.github/workflows/run-tests.yml

```
@@ -0,0 +1,36 @@
+name: Run Unit Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: '*'
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10  
+    strategy:
+      matrix:
+        python-version: ["3.9", "3.10"]
+      fail-fast: false  
+
+    steps:
+    - uses: actions/checkout@v3
```

Would recommend using the [latest major version](https://github.com/actions/checkout/releases/tag/v4.2.2) of `checkout` action:
```suggestion
    - uses: actions/checkout@v4
```